tests/unit/crt_tests.cpp

```diff
@@ -0,0 +1,55 @@
+/**
+ *  @file
+ *  @copyright defined in eosio.cdt/LICENSE.txt
+ */
+#include <cstdint>
+#include <string>
+
+#include <eosio/tester.hpp>
+#include <eosio/crt.hpp>
+
+#include <cstdio>
+
+using eosio::cdt::output_stream;
+
+EOSIO_TEST_BEGIN(output_stream_push)
+   std_err.clear();
+   const char* msg = "abc";
+   _prints(msg, eosio::cdt::output_stream_kind::std_err);
+   CHECK_EQUAL(std_err.to_string(), "abc");
+   CHECK_EQUAL(std_err.index(), 3);
+
+   std_err.clear();
+   const char* msg2 = "";
+   _prints(msg2, eosio::cdt::output_stream_kind::std_err);
+   CHECK_EQUAL(std_err.to_string(), "");
+   CHECK_EQUAL(std_err.index(), 0);
+
+   std_err.clear();
+EOSIO_TEST_END
+
+EOSIO_TEST_BEGIN(output_stream_push_overflow)
+   std_err.clear();
+   const auto initial_capacity = std_err.to_string().capacity();
```

Comment:
I think it worth to check `initial_capaticy` as well. if it will be zero test should pass anyway.

Reply:
Initially I thought we can check here hardcoded 4kb size but that might be too much of details.
So this looks resolved now.